File: flutter_app/lib/features/model/offense_type_dict.dart
class OffenseTypeDictModel {
  final int? typeId;
  final String? offenseCode;
  final String? offenseName;
  final String? category;
  final String? description;
  final double? standardFineAmount;
  final double? minFineAmount;
  final double? maxFineAmount;
  final int? deductedPoints;
  final int? detentionDays;
  final int? licenseSuspensionDays;
  final String? severityLevel;
  final String? legalBasis;
  final String? status;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final DateTime? deletedAt;
  final String? remarks;

  const OffenseTypeDictModel({
    this.typeId,
    this.offenseCode,
    this.offenseName,
    this.category,
    this.description,
    this.standardFineAmount,
    this.minFineAmount,
    this.maxFineAmount,
    this.deductedPoints,
    this.detentionDays,
    this.licenseSuspensionDays,
    this.severityLevel,
    this.legalBasis,
    this.status,
    this.createdAt,
    this.updatedAt,
    this.deletedAt,
    this.remarks,
  });

  OffenseTypeDictModel copyWith({
    int? typeId,
    String? offenseCode,
    String? offenseName,
    String? category,
    String? description,
    double? standardFineAmount,
    double? minFineAmount,
    double? maxFineAmount,
    int? deductedPoints,
    int? detentionDays,
    int? licenseSuspensionDays,
    String? severityLevel,
    String? legalBasis,
    String? status,
    DateTime? createdAt,
    DateTime? updatedAt,
    DateTime? deletedAt,
    String? remarks,
  }) {
    return OffenseTypeDictModel(
      typeId: typeId ?? this.typeId,
      offenseCode: offenseCode ?? this.offenseCode,
      offenseName: offenseName ?? this.offenseName,
      category: category ?? this.category,
      description: description ?? this.description,
      standardFineAmount: standardFineAmount ?? this.standardFineAmount,
      minFineAmount: minFineAmount ?? this.minFineAmount,
      maxFineAmount: maxFineAmount ?? this.maxFineAmount,
      deductedPoints: deductedPoints ?? this.deductedPoints,
      detentionDays: detentionDays ?? this.detentionDays,
      licenseSuspensionDays:
          licenseSuspensionDays ?? this.licenseSuspensionDays,
      severityLevel: severityLevel ?? this.severityLevel,
      legalBasis: legalBasis ?? this.legalBasis,
      status: status ?? this.status,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
      deletedAt: deletedAt ?? this.deletedAt,
      remarks: remarks ?? this.remarks,
    );
  }

  factory OffenseTypeDictModel.fromJson(Map<String, dynamic> json) {
    return OffenseTypeDictModel(
      typeId: json['typeId'],
      offenseCode: json['offenseCode'],
      offenseName: json['offenseName'],
      category: json['category'],
      description: json['description'],
      standardFineAmount: _toDouble(json['standardFineAmount']),
      minFineAmount: _toDouble(json['minFineAmount']),
      maxFineAmount: _toDouble(json['maxFineAmount']),
      deductedPoints: json['deductedPoints'],
      detentionDays: json['detentionDays'],
      licenseSuspensionDays: json['licenseSuspensionDays'],
      severityLevel: json['severityLevel'],
      legalBasis: json['legalBasis'],
      status: json['status'],
      createdAt: _parseDateTime(json['createdAt']),
      updatedAt: _parseDateTime(json['updatedAt']),
      deletedAt: _parseDateTime(json['deletedAt']),
      remarks: json['remarks'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'typeId': typeId,
      'offenseCode': offenseCode,
      'offenseName': offenseName,
      'category': category,
      'description': description,
      'standardFineAmount': standardFineAmount,
      'minFineAmount': minFineAmount,
      'maxFineAmount': maxFineAmount,
      'deductedPoints': deductedPoints,
      'detentionDays': detentionDays,
      'licenseSuspensionDays': licenseSuspensionDays,
      'severityLevel': severityLevel,
      'legalBasis': legalBasis,
      'status': status,
      'createdAt': createdAt?.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
      'deletedAt': deletedAt?.toIso8601String(),
      'remarks': remarks,
    };
  }

  static List<OffenseTypeDictModel> listFromJson(List<dynamic> jsonList) {
    return jsonList
        .map((value) => OffenseTypeDictModel.fromJson(value as Map<String, dynamic>))
        .toList();
  }

  static DateTime? _parseDateTime(dynamic value) {
    if (value == null) return null;
    if (value is DateTime) return value;
    if (value is String && value.isNotEmpty) {
      return DateTime.tryParse(value);
    }
    return null;
  }

  static double? _toDouble(dynamic value) {
    if (value == null) return null;
    if (value is num) return value.toDouble();
    if (value is String && value.isNotEmpty) {
      return double.tryParse(value);
    }
    return null;
  }
}

File: backend/src/main/java/com/tutict/finalassignmentbackend/entity/OffenseTypeDict.java
package com.tutict.finalassignmentbackend.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 违法类型字典表实体类，对应数据库表 "offense_type_dict"
 * 用于定义各种违法类型及其对应的处罚标准
 */
@Data
@TableName("offense_type_dict")
public class OffenseTypeDict implements Serializable {
    /**
     * 序列化版本 UID
     */
    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 类型 ID，主键，自动增长
     */
    @TableId(value = "type_id", type = IdType.AUTO)
    private Integer typeId;

    /**
     * 违法代码
     */
    @TableField("offense_code")
    private String offenseCode;

    /**
     * 违法名称
     */
    @TableField("offense_name")
    private String offenseName;

    /**
     * 违法类别 (超速, 闯红灯, 酒驾, 毒驾等)
     */
    @TableField("category")
    private String category;

    /**
     * 违法描述
     */
    @TableField("description")
    private String description;

    /**
     * 标准罚款金额 (元)
     */
    @TableField("standard_fine_amount")
    private BigDecimal standardFineAmount;

    /**
     * 最低罚款金额 (元)
     */
    @TableField("min_fine_amount")
    private BigDecimal minFineAmount;

    /**
     * 最高罚款金额 (元)
     */
    @TableField("max_fine_amount")
    private BigDecimal maxFineAmount;

    /**
     * 扣分 (0-12分)
     */
    @TableField("deducted_points")
    private Integer deductedPoints;

    /**
     * 拘留天数
     */
    @TableField("detention_days")
    private Integer detentionDays;

    /**
     * 吊销驾照天数
     */
    @TableField("license_suspension_days")
    private Integer licenseSuspensionDays;

    /**
     * 严重程度 (Minor, Moderate, Severe, Critical)
     */
    @TableField("severity_level")
    private String severityLevel;

    /**
     * 法律依据
     */
    @TableField("legal_basis")
    private String legalBasis;

    /**
     * 状态 (Active, Inactive)
     */
    @TableField("status")
    private String status;

    /**
     * 创建时间
     */
    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /**
     * 软删除时间戳
     */
    @TableField("deleted_at")
    @TableLogic(value = "null", delval = "now()")
    private LocalDateTime deletedAt;

    /**
     * 备注
     */
    @TableField("remarks")
    private String remarks;
}